src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -43,6 +54,23 @@
      * @return true if it is apparently still alive (or we cannot tell); false if it is believed to not be running
      */
     public static boolean isAlive(VirtualChannel channel, int pid, Launcher launcher) throws IOException, InterruptedException {
+        Boolean working = workingLaunchers.get(launcher);
+        if (working == null) {
+            // Check to see if our logic correctly reports that an unlikely PID is not running.
+            working = !_isAlive(channel, 9999, launcher);
```

Comment:
It would seem to me that you could use some of this trick: https://github.com/bytemanproject/byteman/blob/master/contrib/bmunit/src/org/jboss/byteman/contrib/bmunit/BMUnitConfigState.java#L433-L520 to get the PID of the current process... then isAlive on the current JVMs PID should return `true` and you could use fork...join on a JVM that outputs its own PID to get a PID that does not exist and verify that isAlive returns false for that PID